sequencer/src/main/java/net/consensys/linea/bundles/LineaLimitedBundlePool.java

```diff
@@ -99,6 +108,8 @@ public LineaLimitedBundlePool(
                         .addArgument(cause::name)
                         .log();
                     removeFromBlockIndex(bundle);
+                    transactionBundleRemovedListeners.forEach(
```

Comment:
should we also add bundleRemovedListeners callback in putOrReplace for existing bundles that get removed?  we are replacing by hash identifier, but technically we are removing it when we replace it.  Seems like it would depend on the intent of a remove listener

Reply:
Good catch, I didn't see the `if(... cause.wasEvicted())`, basically I need to notify the listeners for every removal, so I will move this code outside the `if` so it is called also when a bundle is replaced with `putOrReplace`

Reply:
Added unit tests to verify remove notifications are sent in case of a replace